lib/cuckoo/core/database.py

```diff
@@ -809,8 +809,16 @@ def is_relevant_machine_available(self, task: Task) -> bool:
         @return: boolean indicating if a relevant machine is available
         """
         # Are there available machines that match up with a task?
-        task_arch = next((tag.name for tag in task.tags if tag.name in ["x86", "x64"]), "")
-        task_tags = [tag.name for tag in task.tags if tag.name != task_arch]
+        task_tags = []
+        for tag in task.tags:
+            if tag.name in ["x86", "x64"]:
+                task_arch = tag.name
+            else:
+                task_tags.append(tag.name)
```

Comment:
this is a list of non-arch tags to match against machine tags (which shouldn't have arch in them anymore!)